File: Fundamentals_of_Software_Design/1_Basic_skills/Подсчет символов, строк, слов в файле/main.c
#include <stdio.h>
#include <ctype.h>

void count_file_stats(const char *filename, int *char_count, int *line_count, int *word_count) {
    FILE *file = fopen(filename, "r");
    if (!file) {
        printf("Ошибка: файл '%s' не найден.\n", filename);
        return;
    }

    int ch, prev = ' ';
    *char_count = *line_count = *word_count = 0;

    while ((ch = fgetc(file)) != EOF) {
        (*char_count)++;
        if (ch == '\n') (*line_count)++;
        if (isspace(ch) && !isspace(prev)) (*word_count)++;
        prev = ch;
    }

    fclose(file);
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Использование: %s <имя_файла>\n", argv[0]);
        return 1;
    }

    int chars, lines, words;
    count_file_stats(argv[1], &chars, &lines, &words);

    printf("Символов: %d\n", chars);
    printf("Строк: %d\n", lines);
    printf("Слов: %d\n", words);

    return 0;
}

File: Fundamentals_of_Software_Design/1_Basic_skills/Подсчет символов, строк, слов в файле/main.py
import sys

def count_file_stats(filename):
    try:
        with open(filename, 'r', encoding='utf-8') as file:
            char_count = 0
            line_count = 0
            word_count = 0

            for line in file:
                line_count += 1
                char_count += len(line)
                word_count += len([word for word in line.split() if word])

        return char_count, line_count, word_count
    except FileNotFoundError:
        print(f"Ошибка: файл '{filename}' не найден.")
        return None

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Использование: python script.py <имя_файла>")
    else:
        filename = sys.argv[1]
        result = count_file_stats(filename)
        if result:
            chars, lines, words = result
            print(f"Символов: {chars}")
            print(f"Строк: {lines}")
            print(f"Слов: {words}")
